node/as/thrift.js

```diff
@@ -161,12 +161,13 @@ function register(channel, name, opts, handle) {
                     'Could not serialize thrift');
             }
 
-            res.headers.as = 'thrift';
-            res.setOk(thriftRes.ok);
-            res.send(
-                stringifyResult.value.head,
-                stringifyResult.value.body
-            );
+            if (res.setOk(thriftRes.ok)) {
```

Comment:
I don’t think we can rely on consumers to do this. res.send needs a similar guard. Shame that they’ll plow ahead overwriting the headers though.